client/src/app/containers/Game/index.jsx

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import { useInjectReducer, useInjectSaga } from 'redux-injectors';
+
+import Grid from '@material-ui/core/Grid';
+import RefreshIcon from '@material-ui/icons/Refresh';
+import IconButton from '@material-ui/core/IconButton';
+
+import { selectCurrentUser } from '../../../auth/selectors';
+import { gameSaga } from './saga';
+import { sliceKey, reducer, actions } from './slice';
+import { selectCurrentGame } from './selectors';
+import ActionBox from './ActionBox';
+import PlayingCard from './PlayingCard';
+import PlayerHand from './PlayerHand';
+import CribHand from './CribHand';
+import ScoreBoard from './ScoreBoard';
+
+const showCutCard = phase => {
+  return phase !== 'Deal' && phase !== 'BuildCrib' && phase !== 'Cut';
+};
+
+const handForPlayer = (game, myID, position) => {
+  let isFourPlayer =
+    game.teams.length === 2 && game.teams[0].players.length === 2;
+  if (position === 'across') {
+    if (game.teams.length === 3) {
+      let secondPlayerID = game.teams.filter(
+        t => !t.players.some(p => p.id === myID),
+      )[1].players[0].id;
+      return game.hands[secondPlayerID];
+    } else if (isFourPlayer) {
+      let partnerID = game.teams
+        .filter(t => t.players.some(p => p.id === myID))[0]
+        .players.filter(p => p.id !== myID)[0].id;
+      return game.hands[partnerID];
+    }
+    let opponentID = game.teams.filter(
+      t => !t.players.some(p => p.id === myID),
+    )[0].players[0].id;
+    return game.hands[opponentID];
+  } else if (position === 'right') {
+    if (isFourPlayer) {
+      let rightID = game.teams
+        .filter(t => t.players.some(p => p.id !== myID))[0]
+        .players.filter(p => p.id !== myID)[1].id;
+      return game.hands[rightID];
+    }
+    // nothing!
+    return null;
+  } else if (position !== 'left' || !isFourPlayer) {
+    return null;
+  }
+  // position is left
+  let leftID = game.teams
+    .filter(t => t.players.some(p => p.id !== myID))[0]
+    .players.filter(p => p.id !== myID)[0].id;
+  return game.hands[leftID];
+};
+
+const Game = () => {
+  useInjectReducer({ key: sliceKey, reducer: reducer });
+  useInjectSaga({ key: sliceKey, saga: gameSaga });
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const currentUser = useSelector(selectCurrentUser);
+  const activeGame = useSelector(selectCurrentGame);
+
+  // event handlers
+  const onRefreshCurrentGame = id => {
+    dispatch(actions.refreshGame(id, history));
+  };
+
+  return (
```

Comment:
There's a lot of `<Grid></Grid>` going on. Can we break down subcomponents of the grid for readability?